src/components/Map.tsx

```diff
@@ -0,0 +1,114 @@
+import React, {useEffect, useRef} from 'react';
+import { connect } from "react-redux";
+
+import './Map.css';
+import { State } from "../redux/reducers/reducers";
+
+import {LocationState} from "../redux/reducers/location";
+import {PizzaPlaceState} from "../redux/reducers/pizzaPlaces";
+
+import {getCurrentLocation, setLocation} from "../redux/actions/location";
+import {getPizzaPlaces} from "../redux/actions/pizzaPlaces";
+
+import store from "../redux/store";
+import * as L from 'leaflet';
+import classNames from "classnames";
+import { PizzaPlace } from "../redux/models";
+
+
+interface StateProps {
+    readonly location: LocationState;
+    readonly pizzaPlaces: PizzaPlaceState;
+}
+
+const App = ({location, pizzaPlaces}: StateProps) => {
+    const mapRef: any = useRef(null);
+    const markerRef: any = useRef(null);
+
+    // Get current location once on load
+    useEffect(() => {
+      store.dispatch(getCurrentLocation())
+      store.dispatch(getPizzaPlaces())
+  }, [])
+
+    // Initialize the map on first load
+    useEffect(() => {
+        function onMapClick (d: any) {
```

Comment:
It should be possible to find an actual type for this; the warning about "d implicitly has any type" is usually better resolved by making it something other than `any` rather than making the `any` explicit.

Reply:
Yeah, I would agree that you should avoid using `any` as much as possible. It's rare that a function will truly accept _anything_ as an argument (usually the type is constrained in some way). The more you constrain your types the more the compiler will be able to infer and the more it will be able to help you.

I don't know if it's desirable for this project since it's mostly about learning, but on a normal project I would recommend using [react-leaflet](https://github.com/PaulLeCam/react-leaflet) instead of Leaflet directly. Using a React library to wrap Leaflet will make your life easier (at least that's been my experience with `react-leaflet`) and will remove some of the need for `refs` whose use is generally discouraged where possible: https://reactjs.org/docs/refs-and-the-dom.html#when-to-use-refs

Reply:
Just to drive this point home, if you were to change this param's type to Leaflet's mouse event, it would know what it can expect back:
```suggestion
        function onMapClick(d: L.LeafletMouseEvent) {
```
Given the definition of `LeafletMouseEvent`:
```
export interface LeafletMouseEvent extends LeafletEvent {
    latlng: LatLng;
    layerPoint: Point;
    containerPoint: Point;
    originalEvent: MouseEvent;
}
```
if you were to mistype and write `d.latlng.long` for instance, the compiler would yell at you and you wouldn't have to exercise that code path to find out that you'd made a mistake via an error thrown at runtime.

Reply:
Got it, this is now implemented. I definitely do see the value in this, as I've certainly made that mistake in past dealings with Leaflet haha